test/parse.js

```diff
@@ -843,3 +844,46 @@ test('parse()', function (t) {
 
     t.end();
 });
+
+test('parses empty keys', function (t) {
+    emptyTestCases.forEach(function (testCase) {
+        t.test('parses an object with empty string key with ' + testCase.input, function (st) {
+            st.deepEqual(qs.parse(testCase.input, { allowEmptyKeys: true }), testCase.withEmptyKeys);
+            st.deepEqual(qs.parse(testCase.stringifyOutput, { allowEmptyKeys: true }), testCase.withEmptyKeys);
+
+            st.end();
+        });
+
+        t.test('skips empty string key with ' + testCase.input, function (st) {
+            st.deepEqual(
+                qs.parse(testCase.input),
+                testCase.noEmptyKeys
+            );
+
+            st.deepEqual(
+                qs.parse(testCase.input, { allowEmptyKeys: false }),
+                testCase.noEmptyKeys
+            );
+
+            st.end();
+        });
+    });
+
+    t.test('edge case with object/arrays', function (st) {
+        st.deepEqual(
+            qs.parse('[][0]=2&[][1]=3', { allowEmptyKeys: true }),
+            { '': { '': ['2', '3'] } },
+            'array/object conversion',
+            { skip: 'TODO: figure out what this should do' }
+        );
+
+        st.deepEqual(
+            qs.parse('[][0]=2&[][1]=3&[a]=2', { allowEmptyKeys: true }),
+            { '': { '': ['2', '3'], a: '2' } },
+            'array/object conversion',
+            { skip: 'TODO: figure out what this should do' }
```

Comment:
let's discuss these TODOs. what are your thoughts?

Reply:
Since we parse empty keys I assume output should be equal to key-full strings? 
- If we use `a[b][0]=2&a[b][1]=3` the output object is `{ a: { b: [ '2', '3' ] } }`
- when we replace `a` and `b` with empty strings in this object: `{ '': { '': [ '2', '3' ] } }` and query string should be `[][0]=2&[][1]=3` which is `<EMPTY_KEY>[<EMPTY_KEY>][0]=2` 

Current output for `[][0]=2&[][1]=3` without empty keys flag is ` { 0: '2', 1: '3' }`